docs/docker-env-setup.md

```diff
@@ -0,0 +1,95 @@
+```
+git clone https://github.com/moby/moby
+cd moby
+make binary
+cd bundles/binary
+sudo systemctl stop docker
+sudo systemctl stop containerd
+sudo cp ./* /usr/bin/
+```
+
+2. Download nydus-snapshotter release tarball from [the release page](https://github.com/containerd/nydus-snapshotter/releases). 
+```
+# Get the stable version
+TAG=v0.6.0
+wget https://ghproxy.com/https://github.com/containerd/nydus-snapshotter/releases/download/"$TAG"/nydus-snapshotter-"$TAG"-x86_64.tgz
+tar -xzvf nydus-snapshotter-"$TAG"-x86_64.tgz
+sudo install -D -m 755 nydus-snapshotter/containerd-nydus-grpc /usr/local/bin
+
+wget -O /etc/nydus/nydusd-config.json https://raw.githubusercontent.com/containerd/nydus-snapshotter/"$TAG"/misc/snapshotter/nydusd-config.fusedev.json
+wget -O /etc/nydus/config.toml https://raw.githubusercontent.com/containerd/nydus-snapshotter/"$TAG"/misc/snapshotter/config.toml
+```
+
+3. Download nydus image service release tarball from [the release page](https://github.com/dragonflyoss/image-service/releases). 
+```
+# Get the stable version
+TAG=v2.1.4
+wget https://ghproxy.com/https://github.com/dragonflyoss/image-service/releases/download/"$TAG"/nydus-static-"$TAG"-linux-amd64.tgz
+tar -xzvf nydus-static-"$TAG"-linux-amd64.tgz
+sudo install -D -m 755 nydus-static/* /usr/local/bin
+```
+
+4. Enable `containerd-snapshotter` feature and `nydus`snapshotter in Docker. Add the following to docker's configuration file (typically: /etc/docker/daemon.json). 
+```json
+{
+  "features": {
+    "containerd-snapshotter": true
+  },
+  "storage-driver": "nydus"
+}
+```
+
+5. Enable nydus snapshotter in containerd. Add the following configuration to containerd's configuration file (typically: /etc/containerd/config.toml). 
+```toml
+version = 2
+
+# Plug nydus snapshotter into containerd
+[proxy_plugins]
+  [proxy_plugins.nydus]
+    type = "snapshot"
+    address = "/run/containerd-nydus/containerd-nydus-grpc.sock"
+```
+
+6. Install fuse  
+- centos
+```bash
+# centos 7
+sudo yum install fuse
+# centos 8
+sudo dnf install fuse
+
+sudo modprobe fuse
+```
+
+- ubuntu
+```bash
+sudo apt-get install fuse
+sudo modprobe fuse
+```
+
+7. Start nydus-snapshotter and restart containerd and docker 
+```
+# install nydus snapshotter service
+wget -O /etc/systemd/system/nydus-snapshotter.service https://raw.githubusercontent.com/containerd/nydus-snapshotter/v0.6.0/misc/snapshotter/nydus-snapshotter.fusedev.service
+sudo systemctl enable --now nydus-snapshotter
+sudo systemctl restart containerd
+
+sudo sed -i "s/fd:/unix:/g" /lib/systemd/system/docker.service
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+```
+ 8. Run nydus image in docker
+```
+# Start local registry
+sudo docker run -d --restart=always -p 5000:5000 registry
+# Convert Nydus image
+sudo nydusify convert --fs-version 6 --source ubuntu --target localhost:5000/ubuntu-nydus
```

Comment:
--fs-version 6 is the default version, no?

Reply:
Not the default option in versions <= 2.2.

Reply:
do you add a document to main branch?

Reply:
Fixed.